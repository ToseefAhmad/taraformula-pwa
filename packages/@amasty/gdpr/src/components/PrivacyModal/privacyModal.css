.cancelButton {
  display: none;
}

.richContent table td,
.richContent table th {
  text-align: left;
  padding: 0.44rem;
  font-size: 0.875rem;
  line-height: 1.125em;
  border-right: 1px solid rgb(var(--venia-global-color-border));
  border-bottom: 1px solid rgb(var(--venia-global-color-border));
  word-break: break-word;
}

.richContent {
  composes: root from '~@magento/venia-ui/lib/components/RichContent/richContent.module.css';
}

.versionChanged {
  margin-bottom: 0.5rem;
  padding: 1.5rem;
  font-weight: 600;
  background: #ffeaac;
}

.body {
  composes: body from '~@magento/venia-ui/lib/components/Dialog/dialog.module.css';
  max-height: calc(100vh - 55px);
  -webkit-overflow-scrolling: touch;
}

@media (max-width: 960px) {
  .dialog {
    composes: dialog from '~@magento/venia-ui/lib/components/Dialog/dialog.module.css';
    max-width: 100vw;
  }
}

@media (max-width: 768px) {
    .dialog {
        width: 320px;
    }
}
