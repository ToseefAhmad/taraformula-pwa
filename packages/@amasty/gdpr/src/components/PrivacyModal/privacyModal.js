import React from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import { FormattedMessage, useIntl } from 'react-intl';
import defaultClasses from './privacyModal.css';
import RichContent from '@magento/venia-ui/lib/components/RichContent';
import Modal from '../Modal';
import { bool, func, object, string } from 'prop-types';

const PrivacyModal = props => {
  const {
    onCancel,
    onSubmit,
    isOpen,
    content = {},
    location,
    isVersionChanged
  } = props;
  const { formatMessage } = useIntl();
  const classes = mergeClasses(defaultClasses, props.classes);

  if (!content.content) {
    return null;
  }

  const versionChanged = isVersionChanged && (
    <p className={classes.versionChanged}>
      <FormattedMessage
        id={`privacy.versionChanged`}
        defaultMessage={
          'We would like to inform you that our Privacy Policy has been amended. Please, read and accept the new terms.'
        }
      />
    </p>
  );

  return (
    <Modal
      confirmTranslationId={'privacySetting.accept'}
      classes={{
        cancelButton: classes.cancelButton,
        dialog: classes.dialog,
        body: classes.body
      }}
      confirmText="I have read and accept"
      shouldShowButton={location !== 'registration'}
      isOpen={isOpen}
      onCancel={onCancel}
      onSubmit={onSubmit}
      shouldUnmountOnHide={true}
      title={formatMessage({
        id: 'privacySetting.privacyModal',
        defaultMessage: content.title || 'Privacy Policy'
      })}
    >
      {versionChanged}
      <RichContent
        classes={{ root: classes.richContent }}
        html={content.content}
      />
    </Modal>
  );
};

PrivacyModal.propTypes = {
  isOpen: bool,
  onCancel: func,
  onSubmit: func,
  content: object,
  location: string
};

export default PrivacyModal;
