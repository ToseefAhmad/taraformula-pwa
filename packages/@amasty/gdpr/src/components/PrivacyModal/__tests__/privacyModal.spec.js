import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import PrivacyModal from '../privacyModal';

jest.mock('../../Modal/Modal', () => 'Modal');
jest.mock('@magento/venia-ui/lib/components/RichContent', () => 'RichContent');
jest.mock('react-intl', () => ({
  FormattedMessage: props => (
    <div componentName="Formatted Message Component" {...props} />
  ),
  useIntl: jest.fn().mockReturnValue({
    formatMessage: jest
      .fn()
      .mockImplementation(options => options.defaultMessage)
  })
}));

test('renders no content privacyModal', () => {
  const tree = createTestInstance(<PrivacyModal content={{ content: '' }} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders versionChanged false', () => {
  const tree = createTestInstance(
    <PrivacyModal
      isVersionChanged={false}
      content={{ content: '<div>test</div>' }}
    />
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders versionChanged true', () => {
  const tree = createTestInstance(
    <PrivacyModal
      isVersionChanged={true}
      content={{ content: '<div>test</div>' }}
    />
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
