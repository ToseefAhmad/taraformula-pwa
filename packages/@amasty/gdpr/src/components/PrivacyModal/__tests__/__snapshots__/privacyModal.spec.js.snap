// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders no content privacyModal 1`] = `null`;

exports[`renders versionChanged false 1`] = `
<Modal
  classes={
    Object {
      "cancelButton": undefined,
    }
  }
  confirmText="I have read and accept"
  confirmTranslationId="privacySetting.accept"
  shouldShowButton={true}
  shouldUnmountOnHide={true}
  title="Privacy Policy"
>
  <RichContent
    classes={
      Object {
        "root": undefined,
      }
    }
    html="<div>test</div>"
  />
</Modal>
`;

exports[`renders versionChanged true 1`] = `
<Modal
  classes={
    Object {
      "cancelButton": undefined,
    }
  }
  confirmText="I have read and accept"
  confirmTranslationId="privacySetting.accept"
  shouldShowButton={true}
  shouldUnmountOnHide={true}
  title="Privacy Policy"
>
  <p>
    <div
      componentName="Formatted Message Component"
      defaultMessage="We would like to inform you that our Privacy Policy has been amended. Please, read and accept the new terms."
      id="privacy.versionChanged"
    />
  </p>
  <RichContent
    classes={
      Object {
        "root": undefined,
      }
    }
    html="<div>test</div>"
  />
</Modal>
`;
